mautic/mautic
# Fix inability to update Mautic lead fields if the value that is coming from the integration is empty.

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

When a sales user updates a checkbox in the CRM (Salesforce) to go from checked --> unchecked, a text field in Mautic does not update from 1 to 0.

If the sales user checks an unchecked box, the Mautic field updates to a value of 1.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

1. Create a checkbox field in Salesforce
2. Create a text field in Mautic
3. Set up the Salesforce plugin and map those two fields together, pushing from Salesforce to Mautic
4. Update the checkbox field in Salesforce to checked, then to unchecked.
5. A checked box passes a value of 1, but an unchecked box does not pass any value.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a checkbox field in Salesforce
3. Create a text field in Mautic
4. Set up the Salesforce plugin and map those two fields together, pushing from Salesforce to Mautic
5. Update the checkbox field in Salesforce to checked, then to unchecked.
6. A checked box should pass a value of 1 to Mautic, an unchecked box should pass a value of 0.
